src/stream_inserter.h

```diff
@@ -24,22 +30,87 @@ using greptime::v1::GreptimeRequest;
 using greptime::v1::GreptimeResponse;
 using greptime::v1::InsertRequest;
 using greptime::v1::InsertRequests;
+using greptime::v1::RequestHeader;
+
 using grpc::ClientContext;
 using grpc::Status;
 using grpc::Channel;
 
+
+using TableName = std::string;
+using Timestamp = int64_t;
+using FieldName = std::string;
+using FieldVal = float_t;
+
+struct InsertEntry {
+    Timestamp ts;
+    std::unordered_map<TableName, std::unordered_map<FieldName, FieldVal>> tables;
+
+    InsertEntry():ts(-1) {
+        for(auto&[_, fields] : tables) {
+            fields.clear();
+        }
+        tables.clear();
+    }
+
+    InsertEntry(Timestamp ts_):ts(ts_) {
+        for(auto&[_, fields] : tables) {
+            fields.clear();
+        }
+        tables.clear();
+    }
+
+    void add_point(TableName table_name, FieldName field_name, FieldVal field_val) {
+        tables[table_name].emplace(field_name, field_val);
+    }
+
+    void set_ts(Timestamp ts_) {
+        ts = ts_;
+    }
+
+    uint32_t get_point_num();
+
+};
+
+/// 攒批
+using InsertBatch = std::vector<InsertEntry>;
+
+using greptime::v1::Column;
+using greptime::v1::Column_SemanticType;
+using greptime::v1::ColumnDataType;
+using greptime::v1::InsertRequest;
+using greptime::v1::InsertRequests;
+
+struct LineWriter {
+    // 每个 Table 是一个 InsertRequest
+    std::unordered_map<TableName, 
+                       std::tuple<std::unordered_map<FieldName, std::vector<FieldVal>>,
```

Comment:
As a general purpose client, You should use the [`Value` enum defined in greptime-proto](https://github.com/GreptimeTeam/greptime-proto/blob/dbb8e524ca2dd1a6cb7d655fc68dac75e94ae7d9/proto/greptime/v1/row.proto#L40).